objetos/Curso.java

```diff
@@ -1,9 +1,65 @@
 package objetos;
 
 import java.util.ArrayList;
+
+import dados.DadosCursos;
 import objetos.funcionarios.Professor;
 
 public class Curso {
     private String nome;
-    private ArrayList<Professor> professores;
+    private ArrayList<Professor> professores = new ArrayList<>();
+
+    public Curso(String nome, ArrayList<Professor> professores) {
+        setNome(nome);
+        setProfessores(professores);
+        DadosCursos.adicionarCurso(this);
+    }
+
+    public String getNome() {
+        return nome;
+    }
+
+    public void setNome(String nome) {
+        if (nome.isBlank()) {
+            throw new IllegalArgumentException("Nome não pode estar em branco.");
+        }
+        this.nome = nome;
+    }
+
+    public ArrayList<Professor> getProfessores() {
+        return professores;
+    }
+
+    public void setProfessores(ArrayList<Professor> professores) {
+        for (Professor novoProfessor : professores) {
+            if (professorEhCadastrado(novoProfessor)) {
```

Comment:
O método setProfessores() deve seguir um de dois caminhos:
1. Funcionar como um setter, onde toda a lista de professores será substituída pela nova lista

ou

2. Funcionar como uma adição, onde todos os professores da lista passada como parâmetro serão adicionados a lista de professores atual.

Caso seja a 1ª opção, a validação de verificar se algum professor já é cadastrado é redundante.

Caso seja a 2ª opção, a parte de adição à lista deve ser algo parecido com: `this.professores.add(professores)`, ao invés de `this.professores = professores`, e o nome do método pode ser alterado para adicionaProfessores, ou algo parecido.